File: AutoStock.Domain/Entities/PurchaseOrder.cs
using AutoStock.Domain.Enums;

namespace AutoStock.Domain.Entities;

public class PurchaseOrder
{
    public int Id { get; set; }


    public int SupplierId { get; set; }

    public Supplier? Supplier { get; set; }


    public DateTime OrderDate { get; set; }
        = DateTime.UtcNow;


    public PurchaseOrderStatus Status { get; set; }
        = PurchaseOrderStatus.Draft;


    public decimal TotalAmount { get; set; }


    public string? Notes { get; set; }


    public DateTime? ReceivedAt { get; set; }


    public ICollection<PurchaseOrderItem>
        Items
    { get; set; }
            = new List<PurchaseOrderItem>();
}

File: AutoStock.Client/src/types/purchaseOrder.ts
export type PurchaseOrderStatus =
  1 | 2 | 3 | 4;


export interface PurchaseOrderItem {
  id: number;

  carId: number;

  carName: string;

  quantity: number;

  unitCost: number;

  lineTotal: number;
}


export interface PurchaseOrder {
  id: number;

  supplierId: number;

  supplierName: string;

  orderDate: string;

  status: PurchaseOrderStatus;

  totalAmount: number;

  notes: string | null;

  receivedAt: string | null;

  items: PurchaseOrderItem[];
}


export interface CreatePurchaseOrderItem {
  carId: number;

  quantity: number;

  unitCost: number;
}


export interface CreatePurchaseOrderRequest {
  supplierId: number;

  notes?: string | null;

  items: CreatePurchaseOrderItem[];
}


export interface SupplierPurchaseHistory {
    supplierId: number;
  
    supplierName: string;
  
    totalOrders: number;
  
    draftOrders: number;
  
    submittedOrders: number;
  
    receivedOrders: number;
  
    cancelledOrders: number;
  
    totalReceivedUnits: number;
  
    totalReceivedValue: number;
  
    orders: PurchaseOrder[];
  }